### DISCUSSION TITLE
JS SDK suggestion: provide autocancel context in exception message

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#2774] [Q&A]
As I’ve been stress testing https://pockethost.io, I’ve noticed that autocancel exceptions are difficult to debug. It would be helpful if the exception message showed the exact key collision and URL that caused the cancellation.

**Comment by ganigeorgiev** [comment]
We are not currently printing the url as part of the error message, but _it should be there_ (_*actually this may not be true depending when the request cancellation happened_) if you inspect the error object (eg. `console.log(err.url)`).

The request `cancelKey` at the moment is not logged anywhere since it is a PocketBase thing and it is not part of the actual fetch request (the cancellation error is thrown through the fetch's [`AbortController`](https://developer.mozilla.org/en-US/docs/Web/API/AbortController) signal). I've added in my todo to investigate with the planned JS SDK refactoring whether it is possible to pass some meta data when triggering the `AbortController` so that we can reuse it later in the error wrapping.

Also note that the detection whether an error was caused by request cancellation relies on the [`DOMException`](https://developer.mozilla.org/en-US/docs/Web/API/DOMException) which may not be supported on all environments (_eg. the last time I checked it wasn't supported in React Native_) and you'll get a general error.
